nilearn/_utils/__init__.py

```diff
@@ -9,3 +9,5 @@
 from .cache_mixin import CacheMixin
 
 from .logger import _compose_err_msg
+
+from .exceptions import DimensionError
```

Comment:
It would make more sense to always import DimensionError from nilearn.exceptions rather than from nilearn._utils. You'll need to change the imports in the tests accordingly.